json_options.go

```diff
@@ -0,0 +1,88 @@
+package zap
+
+import "time"
+
+// JSONOption is used to set options for a JSON encoder.
+type JSONOption interface {
+	Apply(*jsonEncoder)
+}
+
+// A MessageFormatter defines how to convert a log message into a Field.
+type MessageFormatter func(string) Field
+
+// Apply implements JSONOption.
+func (mf MessageFormatter) Apply(enc *jsonEncoder) {
+	enc.messageF = mf
+}
+
+// MessageKey encodes log messages under the provided key.
+func MessageKey(key string) MessageFormatter {
+	return MessageFormatter(func(msg string) Field {
+		return String(key, msg)
+	})
+}
+
+// NoMessage drops the message field altogether. It's sometimes useful for
+// testing.
+func NoMessage() MessageFormatter {
```

Comment:
We're not using this in any of our tests now, I don't think we should add this to the exported API. It's trivial to implement outside of the library too